app/controllers/admin/volunteers_controller.rb

```diff
@@ -23,21 +23,17 @@ def show
     end
 
     def update
-      if can_manage_volunteers(@conference)
-        if @conference.update_attributes(conference_params)
-          redirect_to(admin_conference_volunteers_info_path(conference_id: params[:conference_id]), notice: 'Volunteering options were successfully updated.')
-        else
-          redirect_to(admin_conference_volunteers_info_path(conference_id: params[:conference_id]), alert: "Volunteering options update failed: #{@conference.errors.full_messages.join '. '}")
-        end
+      if @conference.update_attributes(conference_params)
```

Comment:
1. the can_manage_volunteers function checks about the roles a user had, and not whether or not the user can access the update function of volunteers. Even though the result is currently the same, we should properly implement the checking (See no 2 below).

2. The abilities are set in https://github.com/openSUSE/osem/blob/master/app/models/ability.rb#L184 and we should use load_and_authorize_resource in the beginning of the volunteers controller to properly authorize.

Again a couple of things to be done when volunteering is properly refactored.